File: 02 백준/08 정렬/Python/Prob11931.py

"""
title  : 11931. 수 정렬하기 4 (Silver 5)
time   : 1272ms      628ms
memory : 84752KB     137480KB
"""

import sys

def solution():
    input = sys.stdin.readline

    N = int(input())
    numbers = [int(input()) for _ in range(N)]

    for num in sorted(numbers, reverse=True):
        print(num)

solution()

File: 02 백준/08 정렬/Java/src/Prob11931.java

/**
* title  : 11931. 수 정렬하기 4 (Silver 5)
* time   : 2608ms
* memory : 164672KB
*/

import java.io.*;
import java.util.*;

public class Prob11931 {
    
    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        int N = Integer.parseInt(br.readLine());

        Integer[] number = new Integer[N];
        for (int i = 0; i < N; i++) {
            number[i] = Integer.parseInt(br.readLine());
        }

        Arrays.sort(number, Collections.reverseOrder());

        for(int i = 0; i < (N-1); i++) {
            sb.append(number[i]).append("\n");
        }

        sb.append(number[N-1]);

        System.out.println(sb);
    }
}
